.github/workflows/beam_PostCommit_Java_Jpms_Direct_Java21.yml

```diff
@@ -0,0 +1,97 @@
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: PostCommit Java Jpms Direct Java21
+
+on:
+  schedule:
+  - cron: '0 */6 * * *'
+  workflow_dispatch:
+
+#Setting explicit permissions for the action to avoid the default permissions which are `write-all` in case of pull_request_target event
+permissions:
+  actions: write
+  pull-requests: write
+  checks: write
+  contents: read
+  deployments: read
+  id-token: none
+  issues: write
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.issue.number || github.event.pull_request.head.label || github.sha || github.head_ref || github.ref }}-${{ github.event.schedule || github.event.comment.id || github.event.sender.login}}'
+  cancel-in-progress: true
+
+env:
+  GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GE_ACCESS_TOKEN }}
+  GRADLE_ENTERPRISE_CACHE_USERNAME: ${{ secrets.GE_CACHE_USERNAME }}
+  GRADLE_ENTERPRISE_CACHE_PASSWORD: ${{ secrets.GE_CACHE_PASSWORD }}
+
+jobs:
+  beam_PostCommit_Java_Jpms_Direct_Java21:
+    if: |
+      github.event_name == 'workflow_dispatch' ||
+      github.event_name == 'schedule' ||
+      github.event.comment.body == 'Run Jpms Direct Java 21 PostCommit'
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    timeout-minutes: 240
+    name: ${{ matrix.job_name }} (${{ matrix.job_phrase }})
+    strategy:
+      matrix:
+        job_name: ["beam_PostCommit_Java_Jpms_Direct_Java21"]
+        job_phrase: ["Run Jpms Direct Java 21 PostCommit"]
+    steps:
+    - uses: actions/checkout@v4
+    - name: Setup repository
+      uses: ./.github/actions/setup-action
+      with:
+        comment_phrase: ${{ matrix.job_phrase }}
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        github_job: ${{ matrix.job_name }} (${{ matrix.job_phrase }})
+    - name: Set up Java
+      uses: actions/setup-java@v3.11.0
+      with:
+        distribution: 'temurin'
+        java-version: |
+          21
+          8
+    - name: run PostCommit Java Jpms Direct Java21 script
+      uses: ./.github/actions/gradle-command-self-hosted-action
+      with:
+        gradle-command: :sdks:java:testing:jpms-tests:directRunnerIntegrationTest
+        arguments:
+          -PskipCheckerFramework
+          -PtestJavaVersion=21
+          -Pjava21Home=$JAVA_HOME_21_X64
```

Comment:
tested by running the command `./gradlew :runners:google-cloud-dataflow-java:examplesJavaRunnerV2IntegrationTest -PdisableSpotlessCheck=true -PdisableCheckStyle=true -PskipCheckerFramework -PtestJavaVersion=21 -Pjava21Home=...sdkman/candidates/java/21-zulu` locally:

<img width="1126" alt="image" src="https://github.com/apache/beam/assets/8010435/73a11d83-dba5-4e41-b975-3d58da24a0d6">